app/components/windows/WidgetEditor.vue

```diff
@@ -61,18 +61,22 @@
           </div>
         </div>
 
-        <div class="code-editor hidden" ref="code">
-          <tabs
-            :hideContent="true"
-            className="widget-editor__top-tabs"
-            :tabs="codeTabs"
-            @input="value => updateCodeTab(value)"
-            :value="currentCodeTab"
-          />
-          <div v-for="tab in codeTabs" :key="tab.value">
-            <slot :name="tab.value" v-if="tab.value === currentCodeTab" />
+        <div class="code-editor hidden" ref="code" v-if="loaded">
+          <div v-if="customCodeIsEnabled">
+            <tabs
+              :hideConent="true"
+              className="widget-editor__top-tabs"
+              :tabs="codeTabs"
+              v-model="currentCodeTab"
+              @input="value => updateCodeTab(value)"
+            />
+            <code-editor v-if="currentCodeTab === 'HTML'" key="html" :value="wData" :metadata="{ type: 'html' }"/>
+            <code-editor v-if="currentCodeTab === 'CSS'"  key="css" :value="wData" :metadata="{ type: 'css' }"/>
+            <code-editor v-if="currentCodeTab === 'JS'" key="js" :value="wData" :metadata="{ type: 'js' }"/>
+            <custom-fields-editor  v-if="currentCodeTab === 'customFields'" key="customFields" :value="wData"/>
```

Comment:
i do like handling this here better, but do all of our widgets support custom code?

Reply:
Just checked and realized that `RecentEents` and `TipJar` doesn't. Good catch. I'll add a flag in `apiSettings` property for that